cli/commands/actions_codegen.go

```
@@ -0,0 +1,124 @@
+			}
+			if ec.Config.Version == "1" {
+				return fmt.Errorf("actions commands can be executed only when config version is greater than 1")
+			}
+			if ec.MetadataDir == "" {
+				return fmt.Errorf("actions commands can be executed only when metadata_dir is set in config")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.actions = args
+			return opts.run()
+		},
+	}
+	f := actionsCodegenCmd.Flags()
+
+	f.StringVar(&opts.deriveFrom, "derive-from", "", "derive action from a Hasura operation")
+
+	f.String("endpoint", "", "http(s) endpoint for Hasura GraphQL Engine")
+	f.String("admin-secret", "", "admin secret for Hasura GraphQL Engine")
+	f.String("access-key", "", "access key for Hasura GraphQL Engine")
+	f.MarkDeprecated("access-key", "use --admin-secret instead")
+
+	// need to create a new viper because https://github.com/spf13/viper/issues/233
+	v.BindPFlag("endpoint", f.Lookup("endpoint"))
+	v.BindPFlag("admin_secret", f.Lookup("admin-secret"))
+	v.BindPFlag("access_key", f.Lookup("access-key"))
+
+	return actionsCodegenCmd
+
+}
+
+type actionsCodegenOptions struct {
+	EC         *cli.ExecutionContext
+	actions    []string
+	deriveFrom string
+}
+
+func (o *actionsCodegenOptions) run() (err error) {
+	migrateDrv, err := newMigrate(o.EC, true)
+	if err != nil {
+		return err
+	}
+
+	actionCfg := actions.New(o.EC, o.EC.MetadataDir)
+
+	var derivePayload actions.DerivePayload
+	if o.deriveFrom != "" {
+		derivePayload.Operation = strings.TrimSpace(o.deriveFrom)
+		o.EC.Spin("Deriving a Hasura mutation...")
+		introSchema, err := migrateDrv.GetIntroSpectionSchema()
+		if err != nil {
+			return err
+		}
+		derivePayload.IntrospectionSchema = introSchema
+		o.EC.Spinner.Stop()
+	}
+
+	if o.EC.Config.Action.Codegen.Framework == "" {
+		infoMsg := fmt.Sprintf(`Could not find codegen config in config.yaml. For setting codegen config, run:
+
+  hasura actions use-codegen`)
+		o.EC.Logger.Error(infoMsg)
+		return nil
```

return an error instead of logging the message